src/frontend/screens/MapScreen/MapScreen.tsx

```diff
@@ -0,0 +1,96 @@
+import * as React from "react";
+import { View } from "react-native";
+import debug from "debug";
+
+import Text from "../../sharedComponents/Text";
+import MapView from "../../sharedComponents/Map/MapView";
+import Loading from "../../sharedComponents/Loading";
+import { useDraftObservation } from "../../hooks/useDraftObservation";
+import { useMapStyle } from "../../hooks/useMapStyle";
+import ObservationsContext from "../../context/ObservationsContext";
+import LocationContext from "../../context/LocationContext";
+import { AddButton } from "../../sharedComponents/AddButton";
+import type { NavigationProp } from "../../types";
+import { BGMapSelector } from "./BGMapSelector";
+import IconButton from "../../sharedComponents/IconButton";
+import MaterialIcon from "react-native-vector-icons/MaterialIcons";
+import { MEDIUM_GREY } from "../../lib/styles";
+import { BottomSheetMethods } from "@gorhom/bottom-sheet/lib/typescript/types";
+import { useExperiments } from "../../hooks/useExperiments";
+const log = debug("mapeo:MapScreen");
+
+interface MapScreenProps {
+  navigation: NavigationProp;
+}
+
+export const MapScreen = ({ navigation }: MapScreenProps) => {
+  const [, { newDraft }] = useDraftObservation();
+  const { styleType, styleUrl } = useMapStyle();
+
+  const [experiments] = useExperiments();
+
+  const sheetRef = React.useRef<BottomSheetMethods>(null);
+
+  const [{ observations, status }] = React.useContext(ObservationsContext);
+  const location = React.useContext(LocationContext);
+
+  const handleObservationPress = React.useCallback(
+    (observationId: string) =>
+      navigation.navigate("Observation", { observationId }),
+    [navigation]
+  );
+
+  const handleAddPress = React.useCallback(() => {
+    log("pressed add button");
+    newDraft(undefined, undefined);
```

Comment:
I changed it to TS, and TS was complaining. I originally changed it to null, which broke the e2e. Changing it to undefined seem to fix it, BUT i can change the `useDraftObservation()` hook to accomodate for that if we are worried

Reply:
Yeah just had a look at the code and this implementation is ok. May as well just write `newDraft()` rather than pass arguments. The Typescript in the hook is a bit weird and could do with a clean up (the reason this broke TS was because while the implementation here https://github.com/digidem/mapeo-mobile/blob/develop/src/frontend/hooks/useDraftObservation.ts#L170 correctly has the ClientGeneratedObservation type (which would allow `{ tags: {} }`), the `newDraft` definition here https://github.com/digidem/mapeo-mobile/blob/develop/src/frontend/hooks/useDraftObservation.ts#L64 does not. Also I don't think the type should include `null`, which means that the default value of `{ tags: {} }` is not set, which is why the e2e broke. Anyway, we can fix up the types in `useDraftObservation` later.